source/agora/node/FullNode.d

```diff
@@ -733,4 +737,29 @@ public class FullNode : API
             });
         }
     }
+
+    /***************************************************************************
+
+        Params:
+            tx_hashes = A Set of Transaction hashes
+
+        Returns:
+            Transactions corresponding to the requested hashes or
+            Transaction.init for hashes that can't be found in the pool
+
+    ***************************************************************************/
+
+    public Transaction[] getTransactions (Set!Hash tx_hashes) @safe
+    {
+        Transaction[] found_txs;
+        foreach (hash; tx_hashes)
+        {
+            if (found_txs.length >= MaxBatchTranscationsSent)
+                break;
+            auto tx = this.pool.getTransactionByHash(hash);
```

Comment:
I feel like we can optimize this in the future. I am pretty sure databases are well optimized for looking up several keys at once, and not just one at a time.

Reply:
Agreed, this was just to get stuff working.